### DISCUSSION TITLE
Ökofen V3.10D - No data available

### DISCUSSION THREAD

**Discussion by denmann67** [dominikamann/oekofen-pellematic-compact#95] [Q&A] [answered]
Hello
I'm having a problem with the integration of Oekofen in home assistant :
Error message :

Cette erreur provient d'une intégration personnalisée

Enregistreur: custom_components.oekofen_pellematic_compact
Source: custom_components/oekofen_pellematic_compact/__init__.py:143
intégration: Ökofen Pellematic Compact (documentation, problèmes)
S'est produit pour la première fois: 10:55:01 (3 occurrences)
Dernier enregistrement: 10:56:01

Error reading pellematic data
Traceback (most recent call last):
  File "/config/custom_components/oekofen_pellematic_compact/__init__.py", line 143, in async_refresh_api_data
    update_result = await self.fetch_pellematic_data()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/oekofen_pellematic_compact/__init__.py", line 159, in fetch_pellematic_data
    result = await self._hass.async_add_executor_job(fetch_data, self._host)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/oekofen_pellematic_compact/__init__.py", line 176, in fetch_data
    str_response = response.read().decode(_CHARSET, "ignore")
                   ~~~~~~~~~~~~~^^
  File "/usr/local/lib/python3.13/http/client.py", line 495, in read
    s = self._safe_read(self.length)
  File "/usr/local/lib/python3.13/http/client.py", line 642, in _safe_read
    raise IncompleteRead(data, amt-len(data))
http.client.IncompleteRead: IncompleteRead(0 bytes read, 5287 more expected)

Please help me to solve this problem.
Thanks in advance

**Comment by dominikamann** [comment]
Hi, what happens if you open your configured url in Browser?

**Comment by denmann67** [comment]
Hallo Dominik Amana,
danke, dass Sie auf meinen Hilferuf geantwortet haben . Ich antworte auf Deutsch, das ist einfacher für mich
Meine url ist: http://192.168.1.***:4321/hdenis/all. Wenn ich das in den Browser (Firefox)eingebe , passiert nichts 
Und hier in HA :
![Screenshot_1](https://github.com/user-attachments/assets/b9e1a00a-44f2-4cdf-a3e4-6bdae1661970)

**Comment by dominikamann** [comment]
Das sieht soweit soweit sehr gut. Was wurde denn beim hinzufügen der Pelletheizung in HA konfiguriert?

**Reply by denmann67** [reply]
Hallo,
Ich habe noch einmal von vorne angefangen, den Port und das Passwort geändert und immer noch die gleiche Fehlermeldung erhalten!
![Screenshot_2](https://github.com/user-attachments/assets/fc159207-2af1-4dc1-9104-abd42661b51b)

**Reply by dominikamann** [reply]
Das sieht ebenfalls gut aus. Ist den auf der Heizung selber der Kompatibilitätsmodus aktiviert? Das könnte noch eine Fehlerquelle sein. Dieser sollte deaktiviert sein. Welche Softwareversion ist den installiert? (auf der Heizung)

**Reply by denmann67** [reply]
Hallo, 
Ich habe verschiedene Kombinationen am Kessel ausprobiert, aber ich kann die Daten nicht in HA hochladen.
Ich habe ein Update installiert, das auf dem Kessel verfügbar war, und dabei eine Fehlermeldung festgestellt
 [
![Screenshot_3](https://github.com/user-attachments/assets/418a9031-e50c-4b84-b4be-dfd5269805c3)
![Screenshot_2](https://github.com/user-attachments/assets/2992f07e-d876-45fb-bdd8-016b5fded3b9)
![Screenshot_5](https://github.com/user-attachments/assets/d6430b83-8785-4402-b418-826f6852d192)
![Screenshot_4](https://github.com/user-attachments/assets/916860d2-5a32-428a-b970-1742a851b64e)
![Screenshot_1](https://github.com/user-attachments/assets/25e37cdd-4354-45d6-af6d-6cad710eb719)
](url)

**Reply by dominikamann** [reply]
Modus TCP muss nicht für die Integration aktiviert werden. Es wird ausschließlich das JSON Interface verwendet.

**Comment by dominikamann** [comment]
![IMG_6748](https://github.com/user-attachments/assets/8d377f16-51c3-48a9-8209-336c21eb38eb)
![IMG_6749](https://github.com/user-attachments/assets/2cf594bc-e3fa-4c6f-8d37-dd0afd27835d)

mir ist zudem aufgefallen dass der Port und das Passwort nicht übereinstimmt zu dem funktionieren Beispiel (URL direkt aus HA aufgerufen)

**Comment by denmann67** [comment]
Zunächst einmal ein kleiner Rückblick, Frohe Weihnachten mit etwas Verspätung
ich muss Sie leider weiter belästigen !!
Ich habe die Verbindungsdaten neu eingegeben und einen neuen Versuch gestartet. Aber es funktioniert nicht
![Screenshot_1](https://github.com/user-attachments/assets/3fea5991-e5bb-47bf-8c03-3f11a031b86c)
![IMG_1026-](https://github.com/user-attachments/assets/cf89ef7f-4c37-4bc5-86f1-990f0dc89b81)

**Comment by dominikamann** [comment]
Frohe Weihnachten 🎄! Ich kann den Fehler leider nicht nachvollziehen. 

Das ist meine Konfiguration:

![image](https://github.com/user-attachments/assets/104c1eca-5919-4ba5-b032-833ba94806bd)

![image](https://github.com/user-attachments/assets/cd6763af-2d4f-47bb-ba7f-8d27625c4255)

Rufe ich http://192.168.178.91:4321/8n2L/all (http://**ip**:**port**/**password**/all) im Browser auf sieht es so aus: 

![image](https://github.com/user-attachments/assets/1809643c-4a03-4a37-bfb9-6fadaabea40f)

Bitte genau das nochmal mit Ihren Daten ausprobieren. Achtung das muss alles im selben Netzwerk passieren. 

Interessant wäre auch ob eventuell eine andere Software Version auf Ihrer Heizung installiert ist und wenn ja welche? Ganz alte Versionen könnten gg. nicht funktionieren. Alle neueren gehen problemlos. 

Grüße

**Comment by denmann67** [comment]
Bei mir 
![Capture d’écran   2024-12-27 à 14 50 57](https://github.com/user-attachments/assets/f5d15168-57fd-4c60-aff8-3f157dbf53ca)
Rufe ich: http://192.168.1.171:4321/dhtX/all passiert nichts.
Meine HA version :
![Screenshot_2](https://github.com/user-attachments/assets/a1d9f637-49d5-419c-b47a-444db4ecdc3a)

Meine Frage : kann man eine neue Version herunterladen und wenn ja, wo finde ich sie ?

**Reply by dominikamann** [reply, marked as answer]
Ich bin mir ziemlich sicher das es an der alten Pellematic Version v3 liegt. Diese hat laut diversen Diskussionen hier einige Probleme nicht nur beim Interface. Der Ökofen Kundendienst kann ein Update durchführen. Hier gibt es aber auch einige Integrations-User die es selbst gemacht haben. Einfach mal in den Diskussionen nach "Update" suchen.

**Reply by dominikamann** [reply]
See https://github.com/dominikamann/oekofen-pellematic-compact/discussions/23#discussioncomment-9015235

**Comment by denmann67** [comment]
vielen Dank für Ihre Intervention. 
Ich denke, ich werde mich direkt mit Oekofen in Verbindung setzen. 
Ich werde in der Zwischenzeit auch Diskussionen nach "Update" suchen
Sobald ich etwas Neues habe werde ich es Ihnen mitteilen.
Bis bald 
diesmal so mit etwas Vorsprung, Gutes und glückliches neues Jahr !!

**Comment by denmann67** [comment]
Hallo 
ich wünsche Ihnen noch ein glückliches, frohes Jahr 2025 !!. 
Überraschung!!! ich habe meinen HA auf die Version 2025.10 aktualisiert und siehe da, die Integration funktioniert!!!
Es gab wohl einen Fehler in der alten Version.
Schönen Abend noch
Denis

**Reply by dominikamann** [reply]
Vielen Dank das wünsche ich ihnen auch! Das freut mich zu hören! Viel Spaß mit der Integration 🚀👍
